src/main/java/com/playmonumenta/networkrelay/RemotePlayerManager.java

```diff
@@ -108,100 +113,105 @@ public void broadcast() {
 	private static final Map<String, RemotePlayer> mRemotePlayersByName = new ConcurrentSkipListMap<>();
 	private static final Set<UUID> mVisiblePlayers = new ConcurrentSkipListSet<>();
 
-	protected RemotePlayerManager() {
+	private RemotePlayerManager() {
 		INSTANCE = this;
 		String lShard = getShardName();
 		try {
 			for (String shard : NetworkRelayAPI.getOnlineShardNames()) {
 				if (shard.equals(lShard)) {
 					continue;
 				}
+				MMLog.fine("Registering shard " + shard);
 				mRemotePlayerShardMapped.put(shard, new ConcurrentSkipListMap<>());
 			}
 		} catch (Exception ex) {
-			CustomLogger.getInstance().ifPresent((logger) -> logger.warning("Failed to get load shards"));
+			MMLog.severe("Failed to get remote shard names");
+			throw new RuntimeException("Failed to get remote shard names");
 		}
 
 		try {
-			NetworkRelayAPI.sendBroadcastMessage(REMOTE_PLAYER_REFRESH_CHANNEL, new JsonObject());
-		} catch (Exception e) {
-			CustomLogger.getInstance().ifPresent((logger) -> logger.warning("Failed to broadcast to channel " + REMOTE_PLAYER_REFRESH_CHANNEL));
+			NetworkRelayAPI.sendExpiringBroadcastMessage(REMOTE_PLAYER_REFRESH_CHANNEL,
+				new JsonObject(),
+				REMOTE_PLAYER_MESSAGE_TTL);
+		} catch (Exception ex) {
+			MMLog.severe("Failed to broadcast to channel " + REMOTE_PLAYER_REFRESH_CHANNEL);
 		}
 	}
 
-	@Nullable
-	protected static RemotePlayerManager getInstance() {
+	public static RemotePlayerManager getInstance() {
 		if (INSTANCE == null) {
-			return null;
+			INSTANCE = new RemotePlayerManager();
 		}
 		return INSTANCE;
 	}
 
-	@Nullable
 	protected static String getShardName() {
 		@Nullable String shardName = null;
 		try {
 			shardName = NetworkRelayAPI.getShardName();
 		} catch (Exception e) {
-			CustomLogger.getInstance().ifPresent((logger) -> logger.severe("Failed to retrieve shard name"));
+			MMLog.severe("Failed to get shard name");
+		}
+		if (shardName == null) {
+			throw new RuntimeException("Got null shard name");
 		}
 		return shardName;
 	}
 
-	protected static Set<String> getAllOnlinePlayersName(boolean visibleOnly) {
+	public static Set<String> getAllOnlinePlayersName(boolean visibleOnly) {
```

Comment:
This was originally for all online players, regardless of visibility, with a separate method for only visible players that was kept below and used internally. I'd keep this as it was before, but I figure requiring a visibility parameter makes it more clear for people using the API.